playbooks/provisioning/openstack/sample-inventory/group_vars/all.yml

```diff
@@ -14,13 +14,21 @@ public_dns_nameservers: []
 
 openstack_ssh_public_key: "openshift"
 openstack_external_network_name: "public"
+# # A Neutron network name for control/management/data (user workloads)
 #openstack_private_network_name:  "openshift-ansible-{{ stack_name }}-net"
-
-## If you want to use a provider network, set its name here.
-## NOTE: the `openstack_external_network_name` and
-## `openstack_private_network_name` options will be ignored when using a
-## provider network.
+# # A dedicated Neutron network name for data (user workloads)
+# # NOTE: this is only supported with Flannel SDN yet
+#openstack_private_data_network_name: "openshift-ansible-{{ stack_name }}-data-net"
+
+# # If you want to use a provider network, set its name here.
+# # NOTE: the `openstack_external_network_name`, `openstack_private_network_name`
+# # and `openstack_private_data_network_name` options will be ignored when using a
+# # provider network.
+# # A provider network name for control/management/data (user workloads)
 #openstack_provider_network_name: "provider"
+# # A dedicated provider network name for data (user workloads)
+# # NOTE: this is only supported with Flannel SDN yet
+#openstack_provider_data_network_name: "provider-data"
```

Comment:
the latter

Reply:
it is just  here to extend the `openstack_provider_network_name` cases, I'm not sure if and how it should be used later

Reply:
I'm confused. Shall the `openstack_private_data_network_name` neutron subnet be created even if we want the provider network? Or shall we omit separation of the data network for that case?

Reply:
You did the right thing I think -- when using a provider network, it may not be possible to create any other networks or subnets, so we should just not allow flannel there.